server/follow/FollowService.js

```diff
@@ -0,0 +1,57 @@
+'use strict'
+
+function execRedis (redisClient, method, args) {
+  return new Promise(function (resolve, reject) {
+    args.push(function (err, result) {
+      if (err) return reject(err)
+      resolve(result)
+    })
+    redisClient[method].apply(redisClient, args)
+  })
+}
```

Comment:
I would prefer this to be an async function rather than using promises

Reply:
I don't understand what you mean with "async". I don't know how to do this better.

The redis client doesn't support promises natively. In its readme, they suggest to use bluebird to transform the APIs. I don't like that solution

I don't like mixing behaviour too: the new way of writing the asynchronous job in es7 is used in this project. The redisClient doesn't support that style an I choose to wrap it as lower as I can.

Any ideas to do this well?